lib/services/s3.js

```diff
@@ -436,13 +459,41 @@ AWS.util.update(AWS.S3.prototype, {
     }
   },
 
+  /**
+   * @api private
+   * @param request
+   */
+  disableBodySigning: function disableBodySigning(request) {
+    // Only file streams are supported (content-md5 still calculated)
+    request.service.validateStreamBody(request.httpRequest.body || '');
+    var headers = request.httpRequest.headers;
+    // Add the header to anything that isn't a presigned url, unless that presigned url had a body defined
+    if (headers.hasOwnProperty('X-Amz-Content-Sha256') || !headers.hasOwnProperty('presigned-expires')) {
+      headers['X-Amz-Content-Sha256'] = 'UNSIGNED-PAYLOAD';
```

Comment:
I think you're probably right that I don't need to check for the header. I'll remove it and verify my test cases with presigned urls still work.